includes/Core/User_Surveys/Survey_Queue.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\User_Surveys\Survey_Queue
+ *
+ * @package   Google\Site_Kit\Core\User_Surveys
+ * @copyright 2023 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\User_Surveys;
+
+use Google\Site_Kit\Core\Storage\User_Setting\Array_Setting;
+
+/**
+ * Class for handling surveys queue.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Survey_Queue extends Array_Setting {
+
+	const OPTION = 'googlesitekit_survey_queue';
+
+	/**
+	 * Adds a new survey to the queue.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $survey The survey object to add to the queue.
```

Comment:
Let's more fully document the shape of the `$survey` here to at least include the fields that are relevant for the queue. I.e. https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#1-1-parameters-that-are-arrays